mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/ChooseSimDialog.java

```diff
@@ -75,8 +76,12 @@ public void onSimSelected(View view) {
     @OnClick(R.id.btn_confirm)
     public void onConfirmClicked() {
         dismiss();
-        if (getActivity() instanceof LinkBankAccountActivity) {
+        if (getActivity() instanceof LinkBankAccountActivity && (selectedSim ==1 || selectedSim == 2)){
             ((LinkBankAccountActivity) getActivity()).linkBankAccount(selectedSim);
         }
+        else
+        {
+            Toast.makeText(getContext().getApplicationContext(),getString(R.string.choose_a_sim),Toast.LENGTH_SHORT).show();
```

Comment:
No need to use application context here. You can go through this [article](https://blog.mindorks.com/understanding-context-in-android-application-330913e32514) to learn more

Reply:
I have updated this change and Thanks for the resource.